itis.playground/Pages/homework.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,677 @@
+}
+
+class ItisStudent: Student {
+    
+    var healthPoint: Int
+    var damage: Int
+    var slogan: String
+    var isAlive: Bool
+    var name: String
+    
+    init() {
+        self.name = ItisFraction.getNameFraction()
+        self.isAlive = true
+        self.healthPoint = 100
+        self.damage = 5
+        self.slogan = "ИТИС зашибись!"
+        
+    }
+    
+    func causeDamage(anotherStudent: Student) {
+        anotherStudent.getLosses(anotherStudent: self)
+        if (!anotherStudent.isAlive) {
+            print ("\(self.name) killed \(anotherStudent.name) \n \(self.slogan)")
+        }
+        else {
+            print("\(self.name) wounded \(anotherStudent.name) \n \(self.slogan) vs \(anotherStudent.slogan) ")
+        }
+    }
+    
+    func getLosses(anotherStudent: Student) {
+        self.healthPoint -= anotherStudent.damage
+        
+        if self.healthPoint < 0 {
+            self.isAlive = false
+        }
+    }
+
+}
+
+class IvmiitStudent: Student {
+    
+    var healthPoint: Int
+    var damage: Int
+    var slogan: String
+    var isAlive: Bool
+    var name: String
+    
+    init() {
+        self.name = IvmiitFraction.getNameFraction()
+        self.isAlive = true
+        self.healthPoint = 100
+        self.damage = 5
+        self.slogan = "IVMIIT the best!"
+    }
+
+    func causeDamage(anotherStudent: Student) {
+        anotherStudent.getLosses(anotherStudent: self)
+        if (!anotherStudent.isAlive) {
+            print ("\(self.name) killed \(anotherStudent.name) \n \(self.slogan)")
+        }
+        else {
+            print("\(self.name) wounded \(anotherStudent.name) \n \(self.slogan) vs \(anotherStudent.slogan)")
+        }
+    }
+    
+    func getLosses(anotherStudent: Student) {
+        self.healthPoint -= anotherStudent.damage
+        
+        if self.healthPoint < 0 {
+            self.isAlive = false
+        }
+    }
+}
+
+class FizFackStudent: Student {
+   
+    var healthPoint: Int
+    var damage: Int
+    var slogan: String
+    var isAlive: Bool
+    var name: String
+    
+    init() {
+        self.name = FizFackFraction.getNameFraction()
+        self.isAlive = true
+        self.healthPoint = 100
+        self.damage = 5
+        self.slogan = "I love FizFack!"
+    }
+    
+    func causeDamage(anotherStudent: Student) {
+        anotherStudent.getLosses(anotherStudent: self)
+        if (!anotherStudent.isAlive) {
+            print ("\(self.name) killed \(anotherStudent.name) \n \(self.slogan)")
+        }
+        else {
+            print("\(self.name) wounded \(anotherStudent.name) \n \(self.slogan) vs \(anotherStudent.slogan)")
+        }
+    }
+    
+    func getLosses(anotherStudent: Student) {
+        self.healthPoint -= anotherStudent.damage
+        
+        if self.healthPoint < 0 {
+            self.isAlive = false
+        }
+    }
+}
+
+class MechMathStudent: Student {
+   
+    var healthPoint: Int
+    var damage: Int
+    var slogan: String
```

Comment:
```suggestion
    let slogan: String
```
поправь везде где у тебя не изменяемые переменные